docs/spec/lightclient/attacks/evidence-handling.md

```diff
@@ -0,0 +1,250 @@
+
+        case AmnesiaAttack: return isPossibleBlock(trustedBlock, lcaEvidence.ConflictingBlock) and 
+                                   trustedBlock.Header.Round != lcaEvidence.ConflictingBlock.Header.Round
+    } 
+}
+
+func isPossibleBlock(trusted Header, conflicting Header) boolean {
+    return trusted.ValidatorsHash == conflicting.ValidatorsHash and
+           trusted.NextValidatorsHash == conflicting.NextValidatorsHash and
+           trusted.ConsensusHash == conflicting.ConsensusHash and 
+           trusted.AppHash == conflicting.AppHash and 
+           trusted.LastResultsHash == conflicting.LastResultsHash 
+}
+```
+
+
+## Light client attack creation
+
+Given a trusted light block `trusted`, a light node executes the bisection algorithm to verify header 
+`untrusted` at some height `h`. If the bisection algorithm succeeds, then the header `untrusted` is verified. 
+Headers that are downloaded as part of the bisection algorithm are stored in a store and they are also in 
+the verified state. Therefore, after the bisection algorithm successfully terminates we have a trace of 
+the light blocks ([] LightBlock) we obtained from the primary that we call primary trace.
+
+### Primary trace 
+
+The following invariant holds for the primary trace:
+
+- Given a `trusted` light block, target height `h`, and `primary_trace` ([] LightBlock): 
+    *primary_trace[0] == trusted* and *primary_trace[len(primary_trace)-1].Height == h* and 
+    successive light blocks are passing light client verification logic. 
+          
+
+### Witness with a conflicting header
+
+The verified header at height `h` is cross-checked with every witness as part of 
+[detection](https://github.com/informalsystems/tendermint-rs/tree/master/docs/spec/lightclient/detection). 
+If a witness returns the conflicting header at the height `h` the following procedure is executed to verify 
+if the conflicting header comes from the valid trace and if that's the case to create an attack evidence:
+
+
+#### Helper functions
+
+We assume the following helper functions:
+
+```go
+// Returns trace of verified light blocks starting from rootHeight and ending with targetHeight.
+trace(lightStore LightStore, rootHeight int64, targetHeight int64) LightBlock[]
+
+
+
+```
+
+
+```go
+func DetectLightClientAttacks(primary PeerID, 
+                              primary_trace []LightBlock, 
+                              witness PeerID) (LightClientAttackEvidence, LightClientAttackEvidence) {
+    primary_lca_evidence, witness_trace = DetectLightClientAttack(primary_trace, witness)
+    
+    witness_lca_evidence = nil
+    if witness_trace != nil {
+        witness_lca_evidence, _ = DetectLightClientAttack(witness_trace, primary)
+    }
+    return primary_lca_evidence, witness_lca_evidence
+}
+
+func DetectLightClientAttack(trace []LightBlock, peer PeerID) (LightClientAttackEvidence, []LightBlock) {
+
+    trusted = trace[0]
```

Comment:
```suggestion
    trusted = trace[0]
```
I do not see where trusted is later used

Reply:
On the next line, but I inlined it in the new version.